chainer/functions/normalization/l2_norm_squared.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class L2NormSquared(function.Function):
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+
+        type_check.expect(
+            x_type.dtype == np.float32,
+            x_type.ndim >= 2,
+        )
+
+    def forward_cpu(self, inputs):
+        x, = inputs
+        N = x.shape[0]
+        self.norm = np.zeros(N, dtype=np.float32)
+        for i in range(N):
+            self.norm[i] = x[i].dot(x[i])
+        return self.norm,
+
```

Comment:
How about `self.norm = numpy.array([b[i].dot(b[i]) for i in range(len(b))], dtype=numpy.float32)` . It need not to initialize `self.norm`.